stixpy/net/tests/test_client.py

```diff
@@ -44,10 +85,106 @@ def test_client(urlopen, client, http_response, time_range, nfiles):
     assert len(query) == nfiles
 
 
+@pytest.mark.parametrize(
+    "time_range, level, dtype, nfiles",
+    [
+        (("2022-01-01T00:00:00", "2022-01-01T03:00:00"), "L1", a.stix.DataType.sci, 6),
+        (("2022-01-01T00:00:00", "2022-01-01T01:00:00"), "L1", a.stix.DataType.sci, 4),
+        (("2022-01-01T00:30:00", "2022-01-01T01:00:00"), "L1", a.stix.DataType.sci, 4),
+        (("2022-01-01T00:35:00", "2022-01-01T00:45:00"), "L1", a.stix.DataType.sci, 3),
+        (("2022-01-01T00:00:00", "2022-01-01T00:35:00"), "L1", a.stix.DataType.sci, 3),
+        (("2022-01-01T02:45:00", "2022-01-01T05:00:00"), "L1", a.stix.DataType.sci, 1),
+        (("2023-01-01T02:45:00", "2023-01-01T05:00:00"), "L1", a.stix.DataType.sci, 0),
+        (("2022-01-01T00:00:00", "2022-01-01T03:00:00"), "L1", a.stix.DataType.ql, 4),
+        (("2022-01-01T00:00:00", "2022-01-01T03:00:00"), "ANC", a.stix.DataType.asp, 1),
+        (("2022-01-01T00:00:00", "2022-01-02T03:00:00"), "ANC", a.stix.DataType.asp, 2),
+        (("2022-01-01T00:00:00", "2022-01-03T03:00:00"), "ANC", a.stix.DataType.asp, 3),
+        (("2022-01-01T00:00:00", "2022-01-05T03:00:00"), "ANC", a.stix.DataType.asp, 3),
+    ],
+)
+def test_local_client(clientlocal, time_range, level, dtype, nfiles):
+    query = clientlocal.search(
+        a.Time(*time_range),
+        a.Instrument.stix,
+        a.Level(level),
+        dtype,
+    )
+    assert len(query) == nfiles
+
+
 @pytest.mark.remote_data
 def test_search_date(client):
     res = client.search(a.Time("2020-05-01T00:00", "2020-05-01T23:59"), a.Instrument.stix)
-    assert len(res) == 38
+    assert len(res) == 39
+    # this might need fixed when we change to ANC to become an level of its own
+
+
+def test_search_max_version(clientlocal):
+    res = clientlocal.search(
+        a.Time("2022-01-01T00:00", "2022-01-01T23:59"), a.Instrument.stix, Version=a.stix.MaxVersion(3)
+    )
```

Comment:
Does it work without the keyword arg e.g.
```
a.Time("2022-01-01T00:00", "2022-01-01T23:59"), a.Instrument.stix, a.stix.MaxVersion(3)
```
if not this could be an issue